api/explorer/api/test/test_ingest.py

```diff
@@ -0,0 +1,199 @@
+        pname="C",
+        version="1.0",
+        broken=False,
+        license="MIT",
+    ),
+    build_inputs=[],
+)
+
+dummy_pkg_b = model.Package(
+    name="B",
+    output_paths={model.OutputPathName.OUT: "/B"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="B", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_c,
+        )
+    ],
+)
+
+dummy_pkg_a = model.Package(
+    name="A",
+    output_paths={model.OutputPathName.OUT: "/A"},
+    nixpkgs_metadata=model.NixpkgsMetadata(
+        pname="A", version="1.0", broken=False, license="MIT"
+    ),
+    build_inputs=[
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_b,
+        ),
+        model.BuildInput(
+            build_input_type=model.BuildInputType.BUILD_INPUT,
+            package=dummy_pkg_d,
+        ),
+    ],
+)
+
+dummy_nix_graph = model.NixGraph(packages=[dummy_pkg_a])
+
+#######################################################################################
+# Fixture(s)
+#######################################################################################
+@pytest.fixture
+def graph_connection():
+    conn = DriverRemoteConnection(
+        "ws://localhost:8182/gremlin",
+        "g",
+        message_serializer=serializer.GraphSONMessageSerializer(),
+    )
+    g = traversal().with_remote(conn)
+    yield conn
+    # Dropping all data from graph on teardown
+    g.V().drop().iterate()
+    g.E().drop().iterate()
+    conn.close()
+
+
+#######################################################################################
+# Tests
+#######################################################################################
+
+
+def test_unit_core_to_graph_model():
+    from explorer.api.ingest import core_to_graph_model
+
+    pkg = model.Package(
+        name="foo",
+        output_paths={
+            model.OutputPathName.DEV: "/foo/dev",
+            model.OutputPathName.LIB: "/foo/lib",
+        },
+        nixpkgs_metadata=model.NixpkgsMetadata(
+            pname="foo", version="1.0", broken=False, license="MIT"
+        ),
+        build_inputs=[],
+    )
+    converted_pkgs = core_to_graph_model(pkg)
+
+    expected_pkg_dev = graph.Package(pname="foo", outputPath="/foo/dev")
+    expected_pkg_lib = graph.Package(pname="foo", outputPath="/foo/dev")
+    assert expected_pkg_dev in converted_pkgs
+    assert expected_pkg_lib in converted_pkgs
+
+
+def test_unit_core_to_graph_model_no_output_path():
+    from explorer.api.ingest import core_to_graph_model
+
+    pkg = model.Package(
+        name="foo",
+        output_paths={},
+        nixpkgs_metadata=model.NixpkgsMetadata(
+            pname="foo", version="1.0", broken=False, license="MIT"
+        ),
+        build_inputs=[],
+    )
+    assert core_to_graph_model(pkg) == []
```

Comment:
Hmm in this case I'm not so sure. `core_to_graph_model` currently outputs `N` Packages, where `N=len(model.pkg.output_paths)`. Since a `model.Package` with `output_paths={}` is a valid object, I kind of feel like returning an empty list here is in line with the data model. What do you think?

Reply:
Ah right, I misread the test. You are right indeed.